src/builtins/ft_cd.c

```diff
@@ -19,9 +19,11 @@ void	ft_cd_update(char *oldpwd, t_data *data)
 	t_list	*oldpwd_env;
 	t_list	*pwd_env;
 	t_env	*content;
-	char	cwd[PATH_MAX];
+	char	*cwd;
+	//char	cwd[PATH_MAX];
 
-	getcwd(cwd, PATH_MAX);
+	//getcwd(cwd, PATH_MAX);
+	cwd = getcwd(NULL, 0);
```

Comment:
I had to change these 2 lines to make the new compilation work. 

The problem was that with the previous definition, the following if statement ```if (oldpwd == NULL || cwd == NULL)``` was always evaluating to true for ```cwd == NULL``` as it had a set size (via PATH_MAX) and hence was occupying some memory and would never be NULL.  cc complained about that as being, I suppose, an arbitrary check.

The new version stores just a pointer to the cwd in the ```char *cwd```, ignoring the buff and size. An actual copy of the cwd will be made via ft_strdup later anyways, so I didn't really see the point in the buffer (plus it needed to be changed to smt bcs cc :D and I'm already using this setup in prompt).